### PR TITLE
Add extract_patches to backend

### PR DESCRIPTION
**Pull request by Dref360**
Should this go to keras_contrib?

### CODE REVIEW

**Comment by Dref360** [comment]
@fchollet Should this go to keras_contrib or this should be on keras itself?
@patyork

**Comment by fchollet** [comment]
Until there is core Keras functionality that uses this function, I don't think we should add it to the backend. We should instead try to minimize the backend's contents to just what is required to run Keras.

**Comment by Dref360** [comment]
The PR is open on keras-contrib. The SSIM loss function will follow after the PR is merged.

https://github.com/farizrahman4u/keras-contrib/pull/11
